src/matcalc/_surface.py

```diff
@@ -0,0 +1,303 @@
+        self,
+        bulk_struct: Structure,
+        symmetrize: bool = True,  # noqa: FBT001,FBT002
+        inplane_supercell: tuple[int, int] = (1, 1),
+        slab_gen_kwargs: dict | None = None,
+        get_slabs_kwargs: dict | None = None,
+    ) -> dict[str, Slab]:
+        """
+        Relax (or single-point calculate) a bulk structure, then generate slab structures
+        using the specified Miller index. Stores the resulting bulk data internally:
+        ``self.final_bulk``, ``self.bulk_energy``, and ``self.n_bulk_atoms``.
+
+        :param bulk_struct: A Pymatgen bulk structure. It is converted to its conventional
+            cell before relaxation or single-point calculation.
+        :type bulk_struct: Structure
+        :param symmetrize: Whether to produce symmetrically distinct slabs. Default True.
+        :type symmetrize: bool, optional
+        :param inplane_supercell: Multiplicative factors for in-plane supercell expansion
+            of the slabs. Default is (1, 1).
+        :type inplane_supercell: tuple[int,int], optional
+        :param slab_gen_kwargs: Additional kwargs passed to :class:`SlabGenerator`.
+            Default is None.
+        :type slab_gen_kwargs: dict | None, optional
+        :param get_slabs_kwargs: Additional kwargs passed to
+            :meth:`SlabGenerator.get_slabs`. Default is None.
+        :type get_slabs_kwargs: dict | None, optional
+        :return: A dictionary of the generated slab structures keyed by names like
+            ``"slab_00"``, ``"slab_01"``.
+        :rtype: dict[str, Slab]
+        """
+        bulk = bulk_struct.to_conventional()
+
+        relax_cell = self.relax_bulk
+        relax_atoms = self.relax_bulk
+
+        relaxer_bulk = RelaxCalc(
+            calculator=self.calculator,
+            fmax=self.fmax,
+            max_steps=self.max_steps,
+            relax_cell=relax_cell,
+            relax_atoms=relax_atoms,
+            optimizer=self.optimizer,
+            **(self.relax_calc_kwargs or {}),
+        )
+        bulk_opt = relaxer_bulk.calc(bulk)
+
+        self.final_bulk = bulk_opt["final_structure"]
+        self.bulk_energy = bulk_opt["energy"]
+        if self.final_bulk is None:
+            raise ValueError("Final bulk structure is None after bulk relaxation.")
+        self.n_bulk_atoms = len(self.final_bulk)
+
+        # Generate slabs
+        slabgen = SlabGenerator(
+            initial_structure=bulk,
+            miller_index=self.miller_index,
+            min_slab_size=self.min_slab_size,
+            min_vacuum_size=self.min_vacuum_size,
+            **(slab_gen_kwargs or {}),
+        )
+        slabs = slabgen.get_slabs(symmetrize=symmetrize, **(get_slabs_kwargs or {}))
+
+        slab_dict = {}
+        for id_, slab in enumerate(slabs):
+            key = f"slab_{id_:02d}"
+            slab_ = slab.make_supercell((*inplane_supercell, 1))
+            slab_dict[key] = slab_
+        return slab_dict
+
+    def calc(
```

Comment:
The reason why I know the miller index is not an init arg is that it is not used in any of the other methods that are defined by the abstract class. The calc method does not use the miller index at all. Only variables that need to be used by calc needs to be in the init.